### PR TITLE
[Debt] Migrate pool candidate test to Playwright

### LINKED ISSUE #10481: ♻️ Move e2e to Playwright - pool-candidate.cy.ts
## ♻️ Debt/Refactor

Time to move our Cypress tests to Playwright
apps/e2e/cypress/e2e/admin/pool-candidate.cy.ts

## ✅ Acceptance Criteria

- [ ] Test moved to Playwright: pool-candidate.cy.ts

## 🛑 Blockers

```[tasklist]
### Blocked By
```

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #10481 

## 👋 Introduction

Migrates the pool candidate test from Cypress to Playwright.

> [!NOTE]
> This also caught and fixed a bug with the expiry date formatting

> [!IMPORTANT]
> This test was actually skipped because it was testing the old page so it does not do exactly what the old test was doing.

## 🧪 Testing

1. Confirm test is passing
2. Confirm test covers a similar case as the previous cypress test

### CODE REVIEW

File: apps/playwright/tests/admin/pool-candidate.spec.ts
Comment (vd1992) [inline apps/playwright/tests/admin/pool-candidate.spec.ts:1]: I'd like a few more assertions to test the page a bit more thoroughly. Such as:
- Verifying the snapshot was loaded, so "Profile snapshot not found." does not appear or asserting snapshot values are present
- Updating candidate notes
- You assert the expiry date after recording final decision, but also assert status appears as "Qualified"
- There are quite a few mutation operations you can chain on that page, maybe after record final decision, test you can then place and then remove

**Review by vd1992 — CHANGES_REQUESTED** [review]

File: apps/playwright/tests/admin/pool-candidate.spec.ts
Comment (esizer) [reply apps/playwright/tests/admin/pool-candidate.spec.ts:1]: This is a simple migration of a test that checked that we could change the status. We were not meant to add additional assertions. Can you create tickets to expand the coverage?

Comment (vd1992) [reply apps/playwright/tests/admin/pool-candidate.spec.ts:1]: Oh, I thought it was a rewrite

Comment (esizer) [reply apps/playwright/tests/admin/pool-candidate.spec.ts:1]: Oh no, we are just trying to get the same coverage from cypress to playwright. I like those assertions though and we should probably get them in eventually!

Comment (vd1992) [reply apps/playwright/tests/admin/pool-candidate.spec.ts:1]: Issue created here https://github.com/GCTC-NTGC/gc-digital-talent/issues/10523

**Review by vd1992 — APPROVED** [review]
Great catch with the date string formatting!

**Pull request merged**